File: backend/src/main/java/ru/vsu/board/model/Comment.java
package ru.vsu.board.model;

import lombok.Data;

import javax.persistence.*;

@Data
@Entity
@Table(name = "comments")
public class Comment {
    @Id
    @SequenceGenerator(name = "commentSeq", sequenceName = "COMMENT_SEQUENCE", allocationSize = 1)
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "commentSeq")
    private Long id;

    @Column(name = "text")
    private String text;

    @ManyToOne
    @JoinColumn(name = "user_id")
    private User user;

    @ManyToOne
    @JoinColumn(name = "post_id")
    private Post post;
}

File: frontend/board/src/app/models/comment.ts
export class Comment {
  public id: number;
  public text: string;
  public username: string;
  public post_id: number;

  constructor(id: number, text: string, username: string, post_id: number) {
    this.id = id;
    this.text = text;
    this.username = username;
    this.post_id = post_id;
  }
}
